File: Builder/java/src/main/java/main/Builder.java
package main;

/**
 * @author Daniel Beard
 *
 */
public class Builder {

	String mLocation, mFirst, mLast;
	int mAge;
	
	public Builder() {
		mAge = -1;
		mLocation = mFirst = mLast = "";
	}
	
	/**
	 * @return the mLocation
	 */
	public String getLocation() {
		return mLocation;
	}
	/**
	 * @param loc the mLocation to set
	 * @return this
	 */
	public Builder Location(String loc) {
		this.mLocation = loc;
		return this;
	}
	/**
	 * @return the mFirst
	 */
	public String getFirst() {
		return mFirst;
	}
	/**
	 * @param firstName the mFirst to set
	 * @return this
	 */
	public Builder First(String firstName) {
		this.mFirst = firstName;
		return this;
	}
	/**
	 * @return the mLast
	 */
	public String getLast() {
		return mLast;
	}
	/**
	 * @param lastName the mLast to set
	 * @return this
	 */
	public Builder Last(String lastName) {
		this.mLast = lastName;
		return this;
	}
	/**
	 * @return the mAge
	 */
	public int getAge() {
		return mAge;
	}
	/**
	 * @param epoc the mAge to set
	 * @return this
	 */
	public Builder Age(int epoc) {
		this.mAge = epoc;
		return this;
	}
	
}

File: Builder/py/src/app/Builder.py
'''
Builder pattern with decision tree

Created on Oct 18, 2019

@author: Daniel Beard
'''

class Builder(object):
    '''
    classdocs
    '''

    def Age(self, params = None):
        self.mAge = params
        return self
     
    def City(self, params = None):
        self.mCity = params
        return self
    
    def State(self, params = None):
        self.mState = params
        return self
    
    def FirstName(self, params = None):
        self.mFirst = params
        return self
     
    def LastName(self, params = None):
        self.mLast = params   
        return self
    
    def getAge(self):
        return str(self.mAge)
     
    def getCity(self):
        return self.mCity
    
    def getState(self):
        return self.mState
    
    def getFirstName(self):
        return self.mFirst
     
    def getLastName(self):
        return self.mLast